core/origin-ipfs/src/decoder.rs

```diff
@@ -12,7 +12,7 @@ pub fn decode_block(cid: Cid, data: Vec<u8>) -> Result<Option<Bytes>> {
                 return Ok(None);
             };
             // Optimistically try to decode the data as unixfs
-            match Data::try_from(data.to_vec().as_ref()) {
+            match Data::try_from(data.to_vec().as_slice()) {
                 Ok(unixfs) => Ok(Some(unixfs.Data.to_vec().into())),
```

Comment:
maybe worth changing the function signature to help with not having to copy the data here.